firewood/tests/db.rs

```diff
@@ -117,48 +107,34 @@ fn test_revisions() {
                     };
                     batch.push(write);
                 }
-                let proposal = db.new_proposal(batch).unwrap();
-                proposal.commit_sync().unwrap();
+                let proposal = Arc::new(db.propose(batch).await.unwrap());
+                proposal.commit().await.unwrap();
```

Comment:
Not for this PR, but we definitely should not require that a `Proposal` be wrapped in an `Arc` here.

Reply:
Let's chat about this. Originally, I wanted `db.propose` to return an `Arc<Proposal>` because you really can't do anything with a Proposal that isn't wrapped in an `Arc`.

Reply:
Documenting offline convo:
- I definitely feel like we should return `Arc<Proposal>` from `propose`
- @richardpringle believes we should leave it since it's easy to change later and we might have a use case for a bare proposal
- I don't see any use case for that, and I'd rather change the API earlier rather than later
- No consensus so it stays the way it is